examples/templates/gcp-devcontainer/main.tf

```diff
@@ -0,0 +1,206 @@
+terraform {
+  required_providers {
+    coder = {
+      source = "coder/coder"
+    }
+    google = {
+      source = "hashicorp/google"
+    }
+  }
+}
+
+provider "coder" {
+}
+
+variable "project_id" {
+  description = "Which Google Compute Project should your workspace live in?"
+}
+
+data "coder_parameter" "zone" {
+  name         = "zone"
+  display_name = "Zone"
+  description  = "Which zone should your workspace live in?"
+  type         = "string"
+  icon         = "/emojis/1f30e.png"
+  default      = "us-central1-a"
+  mutable      = false
+  option {
+    name  = "North America (Northeast)"
+    value = "northamerica-northeast1-a"
+    icon  = "/emojis/1f1fa-1f1f8.png"
+  }
+  option {
+    name  = "North America (Central)"
+    value = "us-central1-a"
+    icon  = "/emojis/1f1fa-1f1f8.png"
+  }
+  option {
+    name  = "North America (West)"
+    value = "us-west2-c"
+    icon  = "/emojis/1f1fa-1f1f8.png"
+  }
+  option {
+    name  = "Europe (West)"
+    value = "europe-west4-b"
+    icon  = "/emojis/1f1ea-1f1fa.png"
+  }
+  option {
+    name  = "South America (East)"
+    value = "southamerica-east1-a"
+    icon  = "/emojis/1f1e7-1f1f7.png"
+  }
+}
+
+provider "google" {
+  zone    = data.coder_parameter.zone.value
+  project = var.project_id
+}
+
+data "google_compute_default_service_account" "default" {
+}
+
+data "coder_workspace" "me" {
+}
+
+resource "google_compute_disk" "root" {
+  name  = "coder-${data.coder_workspace.me.id}-root"
+  type  = "pd-ssd"
+  image = "debian-cloud/debian-12"
```

Comment:
Could we use an image that includes docker pre-installed?

Reply:
I looked into it and an initial search didn't find any official/trust worry images. Most of the images are 3rd party, outdated and charge for usage charges too.